test/DomainController.ts

```diff
@@ -0,0 +1,104 @@
+      " registered");
+  });
+
+  it("should allow the owner to set the domain registration fee", async function () {
+    await domainController.setDomainRegistrationFee(2);
+    const fee = await domainController.domainRegistrationFee();
+
+    expect(fee).to.equal(2);
+  });
+
+  it("should allow the owner to withdraw funds", async function () {
+    await domainController.connect(addr1).registerDomain("com", { value: ethers.parseEther("1") });
+
+    const initialBalance = await ethers.provider.getBalance(owner.address);
+    await domainController.withdrawFunds();
+    const finalBalance = await ethers.provider.getBalance(owner.address);
+
+    expect(finalBalance).gt(initialBalance)
+  });
+
+  it('should register a domain and collect metrics', async () => {
+    const domain = 'example';
+    await domainController.connect(addr1).registerDomain(domain, { value: ethers.parseEther('1') });
+
+    const registeredDomainsCountFilter = domainController.filters.DomainRegistered(null, null, null);
+    const registeredDomainsCountLogs = await domainController.queryFilter(registeredDomainsCountFilter);
+    const registeredDomainsCount = registeredDomainsCountLogs.length;
+    expect(registeredDomainsCount).to.equal(1);
+
+    const registeredDomainsFilter = domainController.filters.DomainRegistered(null, null, null);
+    const registeredDomainsLogs = await domainController.queryFilter(registeredDomainsFilter);
+    const registeredDomains = registeredDomainsLogs.map(log => log.args.domain);
+    expect(registeredDomains).to.eql([domain]);
+  });
+
+  it('should retrieve the list of registered domains for a specific controller sorted by registration date from event', async () => {
+    const domain = 'example';
+    const domain2 = 'example1';
+
+    // Register the domains
+    await domainController.registerDomain(domain, { value: ethers.parseEther('1') });
+    await domainController.registerDomain(domain2, { value: ethers.parseEther('1') });
+
+    const controllerAddress = await owner.getAddress();
+    const controllerDomainsLogs = await domainController.queryFilter(domainController.filters.DomainRegistered(null, controllerAddress, null));
+
+    // Extract the registration dates from the event logs
+    const registrationDates = controllerDomainsLogs.map(log => log.args.registrationDate);
+
+    // Sort the registration dates
+    // By default, sorting is by date, but we can write custom sorting
+    controllerDomainsLogs.sort();
+
+    // Get the domains in the order of registration
+    const controllerDomains = registrationDates.map(date => {
+      const log = controllerDomainsLogs.find(log => log.args.registrationDate === date);
+      return log.args.domain;
+    });
+
+    expect(controllerDomains).to.eql([domain, domain2]);
+  });
+
+  it("should return the list of registered domains sorted by registration date", async function () {
```

Comment:
// line 98
// over logic
// it is better to use event to sort the data
// - 1 point